otherlibs/systhreads/st_stubs.c

```diff
@@ -572,6 +596,9 @@ CAMLexport int caml_c_thread_register(void)
 #endif
   /* Take master lock to protect access to the chaining of threads */
   st_masterlock_acquire(&caml_master_lock);
+  th->uniq_ident = caml_thread_new_ident();
```

Comment:
You're changing `th->uniq_ident` without changing `Ident(th->descr)`. Aren't they supposed to be kept in sync?